### PR TITLE
Python Updates 2022-03-30

### PR DESCRIPTION
**Pull request by mweinelt**
Likely the last python updates series before the upcoming zero hydra failures period.

https://hydra.nixos.org/job/nixpkgs/python-updates/tested#tabs-constituents

The previous cycle was #162538 and is currently still in staging-next due to delays caused by the latest OpenSSL update and problems around x86_64-darwin on hydra.

Latest eval report:
https://malob.github.io/nix-review-tools-reports/nixpkgs:python-updates/nixpkgs_python-updates_1755257.html

### CODE REVIEW

File: pkgs/development/python-modules/ansible-lint/default.nix [hunk 1]
```diff
@@ -18,13 +18,13 @@
 
 buildPythonPackage rec {
   pname = "ansible-lint";
-  version = "5.4.0";
+  version = "6.0.2";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ansible-lint/default.nix:21]: That one was a bit more involved and was already done in https://github.com/NixOS/nixpkgs/pull/166283

File: pkgs/development/python-modules/elasticsearch/default.nix [hunk 2]
```diff
@@ -12,11 +12,11 @@ buildPythonPackage (rec {
   # there's a clear path forward. See
   # https://github.com/elastic/elasticsearch-py/issues/1639 for more
   # info.
-  version = "7.16.3";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/elasticsearch/default.nix:15]: If we follow the comment above we should downgrade to 7.13.X.

File: pkgs/development/python-modules/sagemaker/default.nix [hunk 3]
```diff
@@ -17,14 +17,14 @@
 
 buildPythonPackage rec {
   pname = "sagemaker";
-  version = "2.77.1";
+  version = "2.82.0";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sagemaker/default.nix:20]: I think this is sage related and we should skip updating it.

**Comment by risicle** [comment]
Rebased & dropped the superseded commits

**Comment by risicle** [comment]
Flask-RESTful appears broken due to https://github.com/flask-restful/flask-restful/issues/936. There's a fix, but not merged yet.

**Comment by mweinelt** [comment]
> Flask-RESTful appears broken due to [flask-restful/flask-restful#936](https://github.com/flask-restful/flask-restful/issues/936). There's a fix, but not merged yet.

Adapted a change from flask-restx in [`ee355b0` (#166489)](https://github.com/NixOS/nixpkgs/pull/166489/commits/ee355b0e0ebfd40e352fdd1ab79700f7e2ecc3b8).

**Comment by mweinelt** [comment]
@ofborg eval

**Comment by mweinelt** [comment]
Can we ask upstream about making black tests optional instead of carrying that patch?

**Comment by risicle** [comment]
/me considers writing a no-op dummy `black` package.

**Comment by risicle** [comment]
`mitmproxy` appears broken, will have to investigate tomorrow.

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/105.html

grpc fix over in #167129

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/107.html

**Comment by SuperSandro2000** [comment]
> [shells.darmstadt.ccc.de/~hexa/107.html](https://shells.darmstadt.ccc.de/~hexa/107.html)

Would be nice if we could link this in the first comment otherwise you need to expand multiple times the github thread to be able to see it.

**Comment by mweinelt** [comment]
Git currently fails to build on staging so evals are failing early.

**Comment by mweinelt** [comment]
https://malob.github.io/nix-review-tools-reports/nixpkgs:python-updates/nixpkgs_python-updates_1755257.html

File: pkgs/development/python-modules/pylint/default.nix [hunk 4]
```diff
@@ -74,6 +75,9 @@ buildPythonPackage rec {
   disabledTests = lib.optionals stdenv.isDarwin [
     "test_parallel_execution"
     "test_py3k_jobs_option"
+  ] ++ lib.optionals (pythonAtLeast "3.10") [
+    # https://github.com/PyCQA/pylint/issues/6296
+    "test_functional"
```
Comment (dotlambda) [inline pkgs/development/python-modules/pylint/default.nix:80]: Why don't we add `typing-extensions` to `checkInputs`?

Comment (mweinelt) [reply pkgs/development/python-modules/pylint/default.nix:80]: Yep, I have that already fixed up locally, will be part of the next push.

**Comment by mweinelt** [comment]
Removed pyenchant from pylints check inputs because it introduces a dependency on nuspell/pandoc which is a bit heavy for my taste.

**Pull request merged**